api_tests/src/test_environment.ts

```diff
@@ -0,0 +1,126 @@
+import { Config } from "@jest/types";
+import { LedgerRunner } from "../lib/ledgers/ledger_runner";
+import {
+    execAsync,
+    HarnessGlobal,
+    mkdirAsync,
+    rimrafAsync,
+} from "../lib/utils";
+import NodeEnvironment from "jest-environment-node";
+import { Mutex } from "async-mutex";
+import path from "path";
+
+// ************************ //
+// Setting global variables //
+// ************************ //
+
+export default class E2ETestEnvironment extends NodeEnvironment {
+    private docblockPragmas: Record<string, string>;
+    private projectRoot: string;
+    private testRoot: string;
+    private logDir: string;
+    private ledgerRunner: LedgerRunner;
+    public global: HarnessGlobal;
+
+    constructor(config: Config.ProjectConfig, context: any) {
```

Comment:
```suggestion
    constructor(config: Config.ProjectConfig, context: EnvironmentContext) {
```
Import: `import { EnvironmentContext } from "@jest/environment"`